### PR TITLE
update travis node versions

### PR DESCRIPTION
**Pull request by reqshark**
are we even running anything on the travis ci?

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
@tcr set this up, it looks like it's pulling from his fork: https://travis-ci.org/tcr/node-nanomsg

Can't say I know much about travis-ci.

**Comment by reqshark** [comment]
cool, ya let's use this fork instead

**Comment by nickdesaulniers** [comment]
please update the badge in the readme to:
```markdown
[![Build Status](https://travis-ci.org/nickdesaulniers/node-nanomsg.svg)](https://travis-ci.org/nickdesaulniers/node-nanomsg)
```

**Comment by nickdesaulniers** [comment]
Should we also test older versions of node? or remove 0.10 from testing?  I think we could test all the way back to 0.6, but not have a hard requirement for supporting anything other than 0.12.  That way we can inform users in the readme if they need an explicit version of node.  Once we know, that would be good to put in the readme.

**Comment by reqshark** [comment]
> not have a hard requirement for supporting anything other than 0.12

if we set out a hard requirement for constant support it should be latest `io.js` release. 

but I guess we could also target 0.12 and 0.10, I mean all these versions are the most important areas of coverage IMO, one is probably not necessarily more citical than any other. 

and if 0.6 actually passes then I'll buy you a beer.

**Comment by nickdesaulniers** [comment]
> if we set out a hard requirement for constant support it should be latest io.js release. 

If someone had numbers showing me a significant drop in marketshare of node.js in favor of io.js, then yes I'd be fine with that.  Right now I'm not convinced it's anything more than a flash in the pan power play.  And it's not like we're bending over backwards here to support 0.12 as opposed to io.js.

https://news.ycombinator.com/item?id=9109499

**Comment by reqshark** [comment]
we definitely get better native support in more recent versions, not even considering C++11 features.

for example [i found](https://github.com/reqshark/nanscope) running `nanscope()` impacts the performance of native context switches in `0.11`/`0.12` on methods where invoking the V8 scope handler is unnecessary, something that breaks support for `0.10` completely.

if experimenting with the compiler's taste for node versions is on the table, [let's just drop in some `#include` statements](https://github.com/reqshark/nanomsg.iojs/blob/master/lib/nanomsg.cc#L47-L51)

**Comment by nickdesaulniers** [comment]
What do those headers buy us?  We're not using either?

**Comment by reqshark** [comment]
> What do those headers buy us?

flexibility to exploit different flavors of native support across different node versions.

we get different native implementations of the same export method.

**Comment by reqshark** [comment]
sorry those should have been named `.cc` instead of `.h` but since the compiler doesn't care neither do i

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
+os:
+  - linux
+branches:
+  only:
+    - master
```
Comment (nickdesaulniers) [inline .travis.yml]: will this no longer tests PR's on other branches?

File: package.json [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-  "main": "lib/index.js",
+  "name":               "nanomsg",
+  "version":            "0.4.2",
+  "description":        "Node bindings for nanomsg",
+  "main":               "lib/index.js",
   "dependencies": {
-    "bindings": "1.1.1",
-    "nan": "~1.6.2"
+    "bindings":         "1.2.1",
+    "nan":              "~1.7.0"
```
Comment (nickdesaulniers) [inline package.json]: I'm ok with bumping these, but let's not rewrite the formatting of the entire file.

Comment (reqshark) [inline package.json]: what?! way easier to read this way, check it out in your text editor, you don't agree?

File: .travis.yml (hunk 1 above)
Comment (reqshark) [inline .travis.yml]: no PRs are always tested

if you commit to an open PR travis re-runs each commit

but let's say you were working on a branch and wanted to save some work... if you haven't PRed that yet then travis won't bother running and showing a bunch of extra test outcomes in the travis build area on their site, so we can suppress noise from builds we're not interested in looking at

File: package.json (hunk 2 above)
Comment (nickdesaulniers) [inline package.json]: I'm indifferent; that's subjective and shifts git-blame.

Comment (reqshark) [inline package.json]: ya we should try to preserve empirical history that could explain unknowns down the road. 

so definitely, git-blame is prima facie > subjective discrepancies.  i'll revert that part

**Comment by tcr** [comment]
:+1:  for switching the Travis CI branch to @nickdesaulniers'. I think it required I have collaborator access to this repo, so I used the fork back then.

File: Makefile [hunk 3]
```diff
@@ -5,7 +5,7 @@ ALL:
 	npm i
 
 check:
-	npm t
+	-find test/*.js test/standalone/*.js | xargs -n 1 node | node_modules/tap-spec/bin/cmd.js
```
Comment (nickdesaulniers) [inline Makefile]: woah, what's up with the leading hyphen? is that valid?

Comment (reqshark) [inline Makefile]: to avoid linux edge case using `make`. hyphen suppresses an improper exit code, nice catch

Comment (nickdesaulniers) [inline Makefile]: so it should have the hyphen?

Comment (reqshark) [inline Makefile]: yep. this way travis displays a mocha style spec. 

there might be a better alternative for travis. 

we could run tap-spec as the default `npm test` and have something like `make nyan` for the preferred nyan-cat terminal output (unfortunately not helpful/intended for travis ci output).

**Comment by reqshark** [comment]
landed e6a8e235e6e874d2059bd844a693fddac0bdd31e.

**Pull request closed without merging**